osbs/build/plugins_configuration.py

```diff
@@ -0,0 +1,591 @@
+                            plugin_dict['plugin_args']
+                        )
+                    )
+                except KeyError:
+                    # Malformed config
+                    logger.debug("Invalid custom configuration found for enable_plugins")
+
+    def render_flatpak_create_dockerfile(self):
+        phase = 'prebuild_plugins'
+        plugin = 'flatpak_create_dockerfile'
+
+        if self.pt.has_plugin_conf(phase, plugin):
+            if not self.pt.set_plugin_arg_valid(phase, plugin, 'base_image',
+                                                self.user_params.flatpak_base_image.value):
+                self.pt.remove_plugin(phase, plugin)
+                return
+
+    def render_flatpak_create_oci(self):
+        phase = 'prepublish_plugins'
+        plugin = 'flatpak_create_oci'
+
+        if not self.user_params.flatpak.value:
+            self.pt.remove_plugin(phase, plugin)
+            return
+
+    def render_koji(self):
+        """
+        if there is yum repo user_paramsified, don't pick stuff from koji
+        """
+        phase = 'prebuild_plugins'
+        plugin = 'koji'
+        if not self.pt.has_plugin_conf(phase, plugin):
+            return
+
+        if self.user_params.yum_repourls.value:
+            logger.info("removing koji from request "
+                        "because there is yum repo user_paramsified")
+            self.pt.remove_plugin(phase, plugin)
+        elif self.user_params.flatpak.value:
+            logger.info("removing koji from request "
+                        "because this is a Flatpak built from a module")
+            self.pt.remove_plugin(phase, plugin)
+        elif not self.pt.set_plugin_arg_valid(phase, plugin, "target",
+                                              self.user_params.koji_target.value):
+            logger.info("removing koji from request as not user_paramsified")
+            self.pt.remove_plugin(phase, plugin)
+
+    def render_bump_release(self):
+        """
+        If the bump_release plugin is present, configure it
+        """
+        phase = 'prebuild_plugins'
+        plugin = 'bump_release'
+        if not self.pt.has_plugin_conf(phase, plugin):
+            return
+
+        if self.user_params.release.value:
+            logger.info('removing %s from request as release already user_paramsified',
+                        plugin)
+            self.pt.remove_plugin(phase, plugin)
+            return
+
+        # For flatpak, we want a name-version-release of
+        # <name>-<stream>-<module_build_version>.<n>, where the .<n> makes
+        # sure that the build is unique in Koji
+        if self.user_params.flatpak.value:
+            self.pt.set_plugin_arg(phase, plugin, 'append', True)
+
+    def render_import_image(self, use_auth=None):
+        """
+        Configure the import_image plugin
+        """
+        # import_image is a multi-phase plugin
+        phases = ('postbuild_plugins', 'exit_plugins')
```

Comment:
We only have to handle it being an exit plugin here, so we can simplify this method a little: `phase = 'exit_plugins'`, and no need to loop.

Reply:
Fixed.